notebooks/ndd_example.ipynb

```diff
@@ -0,0 +1,202 @@
+    "                            'query_text': short_query,\n",
+    "                            'k': 100,\n",
+    "                            'model_id': get_model_id(),\n",
+    "                        },\n",
+    "                    },\n",
+    "                },\n",
+    "            ],\n",
+    "        },\n",
+    "    },\n",
+    "    'ext': {\n",
+    "        'generative_qa_parameters': {\n",
+    "            'llm_question': long_query,\n",
+    "            'context_size': 5,\n",
+    "            'llm_model': 'gpt-4',\n",
+    "        },\n",
+    "    },\n",
+    "    'size': 100,\n",
+    "}\n",
+    "do_query(query)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "96ba7b19-4c11-4a00-bada-fe367bb212fd",
+   "metadata": {},
+   "source": [
+    "Next, we add `shingles` to the query and run it again.  There is much more diversity in the retrieved chunks and the generated answer is richer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "d63234b4-fb51-4f07-8033-4e54f634d7da",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[0] Arbitration: Any controversy or claim arising out of or in relation to this Agre\n",
+      "[1] (e) Should an arbitrator refuse or be unable to proceed with arbitration proceed\n",
+      "[2] The American Arbitration Association (\"AAX') shall conduct the arbitration, (b) \n",
+      "[3] (c) as efficient and expeditious a manner as practicable and, in this connection\n",
+      "[4] Any claim or dispute (\"Dispute\") by FIA or Supplier, against the other, or again\n",
+      "[5] The arbitration hearini! shall be held in such neutral location as the parties m\n",
+      "[6] (h) The arbitrator of Arbitration Panel is instructed to schedule promptly all d\n",
+      "[7] or paftW summary judgment is granted, the non-prevailing patty may not raise as \n",
+      "[8] The arbitrator of Arbitration Panel is instructed to schedule promptly all disco\n",
+      "[9] The arbaration hearing shall be held in such neutral location as the parties may\n",
+      "[ANSWER]\n",
+      "The rules of arbitration typically involve a dispute being settled by an arbitrator in accordance with the commercial Arbitration Rules of an organization like the American Arbitration Association [1]. The arbitrator's decision is binding and can be enforced in any court of competent jurisdiction [1]. If an arbitrator is unable or unwilling to proceed, they can be replaced according to the rules of the relevant organization [2]. The arbitration process aims to be efficient and expeditious, with the arbitrator determining the scope of discovery and other procedures [4]. The arbitrator follows existing substantive law and honors any claims of privilege recognized by law [4].\n"
+     ]
+    }
+   ],
+   "source": [
+    "query['_source'].append('shingles')\n",
+    "do_query(query)"
+   ]
+  }
+ ],
```

Comment:
I think you need a cell explaining what changed in the next query, and what happened. This is a pretty complicated JSON, and most people will not know OpenSearch syntax.